### ISSUE TITLE
Undefined local variable or method `test_id' for #<Capybara::Selector::Definition:

### ISSUE THREAD

**Issue by filippoliverani** [teamcapybara/capybara#2443]
**Note:** This is for issues with Capybara.  If you have a howto type question, please ask on the mailing list as requested in the README: http://groups.google.com/group/ruby-capybara

## Meta
Capybara Version:
3.35.0

Driver Information (and browser if relevant):
`:selenium`

## Expected Behavior
Up to version 3.34.0 I was able to use `:test_id` selector without any issue.

## Actual Behavior

After enabling `test_id` with `Capybara.test_id = 'data-test-id'`, calling `click_on('any-test-id')` in an Rspec system test, raises this error:

```
 undefined local variable or method `test_id' for #<Capybara::Selector::Definition:0x0000000115faeab8>
       Did you mean?  test

     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/selector/definition/button.rb:60:in `combine_locators'
     # /usr/local/Cellar/ruby@2.7/2.7.2/lib/ruby/2.7.0/delegate.rb:83:in `method_missing'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/selector/definition/button.rb:12:in `block (2 levels) in <main>'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/selector/selector.rb:63:in `instance_exec'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/selector/selector.rb:63:in `call'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/selector/selector.rb:78:in `expression_for'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/selector/definition/link_or_button.rb:7:in `block (3 levels) in <main>'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/selector/definition/link_or_button.rb:6:in `map'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/selector/definition/link_or_button.rb:6:in `block (2 levels) in <main>'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/selector/selector.rb:63:in `instance_exec'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/selector/selector.rb:63:in `call'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/queries/selector_query.rb:48:in `initialize'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/node/finders.rb:53:in `new'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/node/finders.rb:53:in `find'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/node/actions.rb:26:in `click_link_or_button'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/session.rb:762:in `click_on'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/dsl.rb:53:in `call'
     # /usr/local/lib/ruby/gems/2.7.0/gems/capybara-3.35.0/lib/capybara/dsl.rb:53:in `click_on'
```

## Steps to reproduce

Create an Rspec test with `test_id` enabled and call the `click_on` method.
I think it could be related to this recent change https://github.com/teamcapybara/capybara/commit/455cc503ad849b9f709a97afd61edd5c89f27680#diff-36fca3d679fe3c79eb8a04c3f629646c6a0b90eb7569030d58d71a51c078698dR59

**Comment by geoffharcourt** [comment]
We have the same setting in our app and this caused a very large number of specs in our suite to fail for the same reason.

**Comment by twalpole** [comment]
Fixed in 3.35.1

**Comment by filippoliverani** [comment]
That was fast, thank you! 🙏
